src/main/resources/com/google/api/codegen/go/sample.snip

```diff
@@ -155,12 +164,42 @@
   }
 @end
 
-@private callChain(methodNameComponents)
-  @join resource : methodNameComponents on "."
+@private field(field)
+  @if field.description
+    @join line : util.getDocLines(field.description, 100)
+      // {@line}
+    @end
+
+
+  @end
+  {@field.name} := {@field.defaultValue}  // TODO: Update placeholder value.
+  @if field.example
+    // {@field.example}
+  @end
+@end
+
+@private callChain(class)
+  {@resourceCallChain(class)}({@paramList(class.methodCallFieldVarNames)}){@optionalFieldCallChain(class)}
+@end
+
+@private resourceCallChain(class)
+  @join resource : class.methodNameComponents on "."
     {@resource}
   @end
 @end
 
+@private optionalFieldCallChain(class)
+  @join field : class.optionalFields on EMPTY
+    # Array types must be expanded if set on the request body.
+    # There's not really a good way to do this at the transformer level, so we do it here.
+    @if field.isArray
+      .{@field.setterFuncName}({@field.name}...)
```

Comment:
Do I understand correctly that the client library implements array-field setters (and those alone) as variadics?